parser-typechecker/src/Unison/TermPrinter.hs

```diff
@@ -820,8 +819,10 @@ countName n = let f = \(p, s) -> (s, Map.singleton p 1)
               in PrintAnnotation { usages = Map.fromList $ map f $ splitName n}
 
 splitName :: Name -> [(Prefix, Suffix)]
-splitName n = let ns = splitOn "." (Name.toText n)
-              in dropEnd 1 ((inits ns) `zip` (map dotConcat $ tails ns))
+splitName n =
+  let ns = splitOn "." (Name.toText n)
+  in  filter (not . Text.null . snd) $ inits ns `zip` map dotConcat (tails ns)
+
 -- > splitName "x" == [([], "x")]
 -- > splitName "A.x" == [(["A"], "x")]
 -- > splitName "A.B.x" == [(["A"], "B.x"), (["A.B"], "x")]
```

Comment:
Consider adding these (plus any newly relevant test cases) to a unit test?  I couldn't find an existing one for it.